ghost/job-manager/lib/job-manager.js

```diff
@@ -109,6 +113,13 @@ class JobManager {
                         finished_at: new Date()
                     });
                 }
+            } else {
+                if (typeof message === 'object' && this.#domainEvents) {
+                    // Is this an event?
+                    if (message.event) {
+                        this.#domainEvents.ee.emit(message.event.type, message.event.data);
```

Comment:
`ee` is a private member, so we shouldn't be accessing it here.

I understand the reasoning behind needing to access the raw event emitter here as the js objects aren't serialized with all the info.

It would be really cool if we could either remove the private restriction from the `ee` member, ideally we wouldn't do that though because that opens the surface area of our interface a huge amount, or if we could add another couple of methods.

Perhaps something like:
```typescript
type RawEventData {
  name: string,
  data: object
}
```

```typescript
getRawEventData(event: IEvent): RawEventData
dispatchRaw(rawEvent: RawEventData): void
```

We can even skip the whole `getRawEventData` if you want, but it might be a good idea to keep the whole logic of reading `consturctor.name` in that same class.

Reply:
Thanks, I agree. I've created a dispatchRaw method, but didn't add a new RawEventData interface to keep it a bit more readable. The `dispatch` is updated to call the `dispatchRaw` now

Reply:
Awesome okay 👍